test/syncenginetestutils.h

```diff
@@ -744,10 +755,32 @@ class FakeHangingReply : public QNetworkReply
     qint64 readData(char *, qint64) override { return 0; }
 };
 
+// A delayed reply
+template <class OriginalReply>
+class DelayedReply : public OriginalReply
+{
+public:
+    template <typename... Args>
+    explicit DelayedReply(quint64 delayMS, Args &&... args)
+        : OriginalReply(std::forward<Args>(args)...)
+        , _delayMs(delayMS)
+    {
+    }
+    quint64 _delayMs;
+
+    void respond() override
+    {
+        QTimer::singleShot(_delayMs, static_cast<OriginalReply *>(this), [this] {
+            // Explicit call to bases's respond();
+            this->OriginalReply::respond();
+        });
```

Comment:
Why doesn't this mean the OriginalReply's 0 timer *and* the delayed timer fire?

Reply:
I don't understand the question.

Reply:
The constructor of the underlying class sets up a single shot timer already. Won't this just start a second timer?

Reply:
Harder to override in a generic way.